packages/service-worker/src/update.ts

```diff
@@ -23,13 +24,30 @@ import {ERR_SW_NOT_SUPPORTED, NgswCommChannel, UnrecoverableStateEvent, UpdateAc
  */
 @Injectable()
 export class SwUpdate {
+  /**
+   * Emits a `VersionDetectedEvent` event whenever a new version is detected on the server.
+   *
+   * Emits a `VersionInstallationFailedEvent` event whenever checking for or downloading a new
+   * version fails.
+   *
+   * Emits a `VersionReadyEvent` event whenever a new version has been downloaded and is ready for
+   * activation.
+   */
+  readonly versionUpdates: Observable<VersionEvent>;
+
   /**
    * Emits an `UpdateAvailableEvent` event whenever a new app version is available.
+   *
+   * @deprecated use {@link versionUpdates} instead.
```

Comment:
```suggestion
   * @deprecated Use {@link versionUpdates} instead.
```

Maybe also mention that they can filter to only keep `VersionReadyEvent` (or even show the code to do it).